### PR TITLE
Test: add test for lb pressure tensor acf against viscosity

### PR DESCRIPTION
**Pull request by RudolfWeeber**
Fixes #

Description of changes:
-

### CODE REVIEW

File: testsuite/python/lb_pressure_tensor_acf.py [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+N_CELLS = 12
+TAU = 0.002
+SEED = 1
+DENS = 2.4
+VISC = 1.8
+KT = 0.8
+
+
+class TestLBPressureACF:
+    """Tests that the thermalized LB pressure auto correlation funciton
```
Comment (KaiSzuttor) [inline testsuite/python/lb_pressure_tensor_acf.py]: typo

File: testsuite/python/lb_pressure_tensor_acf.py [hunk 2]
```diff
@@ -0,0 +1,148 @@
...
+        tol_global = 4 / np.sqrt(steps)
+        tol_node = tol_global * np.sqrt(N_CELLS**3)
+
+        # check single node
+        for i in range(3):
+            for j in range(i + 1, 3):
+                avg_ij = np.average(p_node[:, i, j])
+                avg_ji = np.average(p_node[:, i, j])
+                self.assertEqual(avg_ij, avg_ji)
+
```
Comment (KaiSzuttor) [inline testsuite/python/lb_pressure_tensor_acf.py:83]: remove blank

Comment (RudolfWeeber) [reply testsuite/python/lb_pressure_tensor_acf.py:83]: This went through the formatting script. Which blank?

Comment (jngrad) [reply testsuite/python/lb_pressure_tensor_acf.py:83]: line 85 is empty and separates two assertions, this also happens at lines 94, 72, etc. This looks intentional to me, as these blanks seem to group together assertions with their pre-conditions/variables. I don't have strong opinion against it, it just looks less compact compared to other python tests.

**Comment by KaiSzuttor** [comment]
is there a reason why the coverage report is red?

**Comment by jngrad** [comment]
> is there a reason why the coverage report is red?

The codecov service was perturbed this afternoon. At some point it said espresso had no files, even when I logged in with my GitHub credentials. In the last commit of this PR, 3 coverage jobs did nothing with the codecov upload script (logfiles [1](https://gitlab.icp.uni-stuttgart.de/espressomd/espresso/-/jobs/245156), [2](https://gitlab.icp.uni-stuttgart.de/espressomd/espresso/-/jobs/245207), [3](https://gitlab.icp.uni-stuttgart.de/espressomd/espresso/-/jobs/245146)) (it probably got a 500 error string and executed it as bash?), the fourth coverage job got the bash script from codecov but had trouble uploading the data (logfile [4](https://gitlab.icp.uni-stuttgart.de/espressomd/espresso/-/jobs/245147)). I've just restarted the 3 coverage jobs. This was probably not a serious issue, as codecov status didn't report any incident (https://status.codecov.io/).

**Comment by RudolfWeeber** [comment]
One of the runners timed out. Probably, we have to re-run the entire pipeline, once the virtual box issue is fixed.

**Comment by fweik** [comment]
It's fine, I've rerun it, codecov seems to be back also.

**Comment by fweik** [comment]
Should we merge that now, or do you want to wait for a fix?

**Comment by RudolfWeeber** [comment]
I think it can be merged now. It just only tests the GPu code for now.

**Review by fweik — APPROVED** [review]

**Comment by KaiSzuttor** [comment]
btw. this test is not usable for the CPU LB because it takes ages to finish

**Comment by KaiSzuttor** [comment]
``` bash
======================================================================
FAIL: test (__main__.TestLBPressureACFCPU)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "../testsuite/python/lb_pressure_tensor_acf.py", line 84, in test
    self.assertLess(avg_ij, tol_node)
AssertionError: 22.45611059064408 not less than 0.7436128024718242

----------------------------------------------------------------------
Ran 2 tests in 6383.608s
```

**Comment by RudolfWeeber** [comment]
The expectation value of the single node off-diagonal stress element is not zero for LB CPU. That’s the actual underlying bug.

This has to be checked in older Espresso versions. @pkreissl, any news on that?

**Comment by fweik** [comment]
@RudolfWeeber is think this comment should have been on the issue #3804 no?

**Comment by RudolfWeeber** [comment]
Yes. I just pressed the reply button to Kai posting the outcome of the commented out CPU test.

**Comment by KaiSzuttor** [comment]
> The expectation value of the single node off-diagonal stress element is not zero for LB CPU. That’s the actual underlying bug.

I'm aware of that, just wanted to mention that the runtime for the test is not compatible with test-driven bugfixing

**Pull request merged**